Fix several race conditions and sources of test flake

fosterseth opened the pull request:
Fixed up a few places in the code where we do blocking send and receive calls, but don't select on the overall context being finished.

This should help mitigate some race conditions we've discovered in testing when starting up and shutting down nodes.

fosterseth committed:
Add select cases for finished contexts on channel sends and receives

fosterseth commented on pkg/netceptor/netceptor.go:
```diff
@@ -852,16 +852,28 @@ func (s *Netceptor) SubscribeRoutingUpdates() chan map[string]string {
+			}
+		}()
 	}
+	wg.Wait()
```
this wait makes me a little nervous because it blocks the entire flood call until all of the writes are done, but at the same time not waiting on these go-routines also makes me nervous (i.e. go-routine leaks).

@ghjm any opinions on this? I could be convinced to just drop the wg.Wait() here

ghjm replied:
I don't see the need for the WaitGroup at all.  We're not holding any locks and there's no cleanup code at the end.  I don't immediately recall any cases where a caller of flood() would depend on the messages being sent before the function returns, though it might be worthwhile to look at the places it's called and docker check this.

Also, it would be cleaner to just launch the goroutine inside the first loop, which would eliminate the need for the temporary data structures and the second loop.

fosterseth replied:
@ghjm okay I'm going to remove the waitgroup. I didn't have any evidence of this being a problem before, so no need to introduce a waitgroup here. thanks

fosterseth committed:
k8s skip stdin if stdin file is 0 bytes

shanemcd commented:
It looks like this is leaking a whoooole lot of goroutines. Going to clone this locally and see if I can find what's going on.

shanemcd committed:
Always run matrix tests to completion

shanemcd committed:
Remove sockets before archiving test results

Fixes https://github.com/ansible/receptor/issues/508

shanemcd committed:
Only run pull_request check on PRs

shanemcd committed:
Add retry loop to flaky test

shanemcd committed:
Wait for concurrent processes to finish in a safe way

shanemcd committed:
PR Feedback

shanemcd committed:
Dont shut down listener when we cant accept connection

shanemcd committed:
Respect context is websocket session

shanemcd committed:
Wait longer for node to shut down

This is a bandaid for a flaky test

shanemcd committed:
Add some missing locks

shanemcd committed:
Fix data race

b.server was being simultaneously written and read in separate go routines

shanemcd committed:
Dont close sessinon twice.

This caused an error when running the app with -race. This also gets call in the defer in runProtocol.
